src/adam_methods.jl

```diff
@@ -0,0 +1,654 @@
+        y[i+1]=y[i]
+        for j=1:steporder
+            y[i+1] += dt[i]*b_exp[steporder, j]*dy[i-(steporder-1) + (j-1)]
+        end
+
+        ##(E)valuate function F at the approximate point t[i+1],y[i+1]
+        dy[i+1] = F(t[i+1],y[i+1])
+
+        ##(C)orrect the formula using implicit Adam Moulton coefficients
+        y[i+1]=y[i]
+        for j = 1 : steporder
+            y[i+1] += dt[i]*b_imp[steporder, j]*dy[i - (steporder -1) + j]
+        end
+
+        ##(E)valulate the function anew at corrected approximatiion t[i+1],y[i+1]
+        dy[i+1] = F(t[i+1], y[i+1])
+    end
+    return t, y
+end
+
+ode_am5(F::Function,y0, t)= ode_am(F::Function,y0, t,5)
+################################################################################
+# Fixed Step Adams-Bashforth multi-step solver for nonstiff problems
+#
+#     [T,X] = ODE_AB(ODEFUN, X0, TSPAN,ORDER=4) with TSPAN = [T0:H:TFINAL]
+#
+#     Note: If method is run at order k, the first step is taken at order 1,
+#     and the order increases each step until it is order k on the kth step.
+#     Thus results may be of lower accuracy than expected.
+#     (Main ref: Hairer & Wanner 1996, Vol I, p.359-360, 372)
+################################################################################
+
+function ode_ab(F, x0, tspan, order::Integer)
+    h = diff(tspan)
+    x = Array(typeof(x0), length(tspan))
+    x[1] = x0
+
+    if 1 <= order <= 4
+        b = ms_coefficients4
+    else
+        b = zeros(order, order)
+        b[1:4, 1:4] = ms_coefficients4
+        for s = 5:order
+            for j = 0:(s - 1)
+                # Assign in correct order for multiplication below
+                #  (a factor depending on j and s) .* (an integral of a polynomial with -(0:s), except -j, as roots)
+                p_int = polyint(poly(diagm(-[0:j - 1; j + 1:s - 1])))
+                b[s, s - j] = ((-1)^j / factorial(j)
+                               / factorial(s - 1 - j) * polyval(p_int, 1))
+            end
+        end
+    end
+
+    # TODO: use a better data structure here (should be an order-element circ buffer)
+    xdot = similar(x)
+    for i = 1:length(tspan)-1
+        # Need to run the first several steps at reduced order
+        steporder = min(i, order)
+        xdot[i] = F(tspan[i], x[i])
+
+        x[i+1] = x[i]
+        for j = 1:steporder
+            x[i+1] += h[i]*b[steporder, j]*xdot[i-(steporder-1) + (j-1)]
+        end
+    end
+    return vcat(tspan), x
+end
+
+# Use order 4 by default
+ode4ms(F, x0, tspan) = ODE.ode_ab(F, x0, tspan, 4)
+ode5ms(F, x0, tspan) = ODE.ode_ab(F, x0, tspan, 5)
+###############################################################################
+#
+# HELPER FUNCTIONS and TABLEAUS FOR AB SOLVERS
+#
+###############################################################################
+##Adam Bashforth Coefficients for Explicit Method
+const ms_coefficients4 = Rational{Int64}[1        0        0         0
```

Comment:
Since the elements in this matrix are going to be converted to float64 anyway. Maybe it is better to just use a pre-evaluated float64 elements.